maestro-orchestra-models/src/main/java/maestro/orchestra/Commands.kt

```diff
@@ -748,6 +748,22 @@ data class TravelCommand(
 
 }
 
+data class AssertOutgoingRequestsCommand(
+    val url: String? = null,
+    val headersPresent: List<String> = emptyList(),
+    val headersAndValues: Map<String, String> = emptyMap(),
+    val httpMethodIs: String? = null,
+    val requestBodyContains: String? = null,
+) : Command {
+    override fun description(): String {
+        return "Assert outgoing requests to $url"
+    }
+
+    override fun evaluateScripts(jsEngine: JsEngine): Command {
+        return this
```

Comment:
Assertions can have JS resolutions from runScript output I would recommend using evaluate script in case of string.